components/omega/doc/design/Logging.md

```diff
@@ -0,0 +1,109 @@
+<!--- Omega OMEGA logging system Requirements and Design ------------------------------------>
+
+# OMEGA Requirements and Design:
+
+## *OMEGA logging system*
+
+## 1 Overview
+
+The OMEGA logging system offers a standardized Application Programming Interface (API) that facilitates the tracking of events occurring during the runtime of all Omega sub-components. Each event comprises a descriptive message along with an indicator denoting its severity level.
+
+## 2 Requirements
+
+### 2.1 Requirement: Basic logging
+
+It is a requirement for the OMEGA logging system to furnish Omega sub-components with a mechanism to store log messages onto a file system, based on their corresponding severity levels.
+
+### 2.2 Requirement: Inter-operability with E3SM
+
+It is a requirement for the OMEGA logging system to recognize the log root filename defined in E3SM as a part of the input configuration.
+
+### 2.3 Requirement: Multi-processing
+
+It is a requirement that each process should have the capability to generate log messages independently. Additionally, there should be a mechanism in place to control the number of processes generating log messages simultaneously. This functionality is desired to ensure efficient management of system logs and prevent potential system overloading due to excessive logging activities.
+
+### 2.4 Requirement: Multi-threading
+
+It is a requirement for the OMEGA logging system to provide thread-safe logging capabilities. This functionality is necessary to ensure that concurrent logging activities by multiple threads are executed in a synchronized manner, without causing data corruption or inconsistencies in the generated log files.
+
+### 2.5 Requirement: Omega data types
+
+It is a requirement for the OMEGA logging system to support a subset of Omega data types, in order to generate well-formatted log messages that include these data types. This functionality is crucial to ensure that all relevant information is captured in the logs and to facilitate effective debugging and troubleshooting activities.
+
+### 2.6 Requirement: Buffered file input/output
+
+It is a requirement for the OMEGA logging system to be equipped with buffered file IO capabilities, with the aim of reducing the number of time-consuming file IO operations. The API should provide an option that allows the user to choose between buffered and unbuffered file IO modes. This functionality is crucial to enhance the performance and efficiency of the logging system, by minimizing the overhead associated with file IO operations.
```

Comment:
I would just add here that the unbuffered capability is crucial for debugging purposes, where unwritten log messages may be lost, leading to confusion.